python/cudf/dataframe/datetime.py

```diff
@@ -193,14 +194,13 @@ def fillna(self, fill_value, inplace=False):
 
     def sort_by_values(self, ascending=True, na_position="last"):
         sort_inds = get_sorted_inds(self, ascending, na_position)
-        col_keys = cudautils.gather(data=self.data.mem,
-                                    index=sort_inds.data.mem)
+        out_col = cpp_copying.apply_gather_column(self, sort_inds.data.mem)
         mask = None
         if self.mask:
             mask = self._get_mask_as_column()\
                 .take(sort_inds.data.to_gpu_array()).as_mask()
             mask = Buffer(mask)
```

Comment:
Mask is being gathered here as well by the above column gather call, so can just set mask to `out_col.nullmask` if it exists.